docs/metrics.md

```diff
@@ -0,0 +1,238 @@
+also measurement duration of the send. For example in code you can take a look actual
+actual code in [writer.go](../common/event/writer.go) where we are sending multiple
+fields per metric and demonstrate full potential of these metrics.
+
+Previous code example will result in following metrics to be reported:
+
+```
+measurement,tag1=val1 field1=1 [timestamp]
+measurement,tag1=val1 field1_mean=1,field1_median=1,field1_min=1,field1_p10=1,field1_p30=1,field1_p70=1,field1_p90=1,field1_max=1,field1_count=1,field1_poolsize=1 [timestamp]
+```
+
+In following text we will talk about aggregating over time interval
+which is always 1 second.
+
+First metric is self explanatory, but we can see that Histogrammable metric
+reports multiple percentiles, mean, min and max. These values can be the same if
+we don't receive send values during 1 second. Moreover it reports `count`
+and `poolsize`, where `count` describes number of times this metric was sent
+and `poolsize` is internal metric, which will be described later.
+
+## Types and aggregation of metrics
+
+### Metric types
+
+We mentioned in previous part that there are two ways how to send metrics in ECS
+resulting in two different outcomes, based on these outcomes we talk about two
+metric types:
+
+1) Counter - `Send`
+2) Histogrammable - `SendHistogrammable`
+
+Creation of both metrics is done by `NewMetric(measurement, timestamp)`,
+so both use same object `Metric`. The distinction is done by Send methods:
+`Send` and `SendHistogrammable`.
+
+Simple `Send(*Metric)`is used to just store metric with given tags and fields without
+creating any other information. This metric is just aggregated
+(more about it later).
+
+Sending metric through `SendHistogrammable(*Metric)`
+will result in metric being added into the collection of metrics with same
+measurement and tags. This collection is used to compute percentiles
+(10, 30, 50, 70, 90), min, max and count (count of elements added into
+the collection since last reset). These values are reported as fields
+by appending strings to the name of original field:
+
+| Meaning | Appended field  |
+| --------|-------------|
+| 10-percentile | field_10p |
+| 30-percentile | field_30p |
+| 70-percentile | field_70p |
+| 90-percentile | field_90p |
+| median (50-percentile) | field_median |
+| mean | field_mean |
+| min | field_min |
+| max | field_max |
+| count | field_count |
+
+### Aggregation
+
+In order to save network bandwidth and RAM usage we are aggregating all metrics.
+This aggregation is done by grouping metrics per second intervals based on timestamps
+rounded to seconds, measurement name and tags. This aggregating is the same of both
+types of metrics. This means that if you send metrics with
+same measurement name and tags, but different timestamps (unix in nanoseconds)
+but difference under one second all of them will be aggregated into
+one bucket, where timestamp is rounded down at seconds.
```

Comment:
> This means that if you send metrics with
same measurement name and tags, but different timestamps (unix in nanoseconds)
but difference under one second all of them will be aggregated into
one bucket, where timestamp is rounded down at seconds.

I can't understand this sentence, could you rephrase?